### PR TITLE
PMM-9354 external clickhouse DB instance

### PR DESCRIPTION
**Pull request by rasika-chivate**
Ability to use external ClickHouse database instance for PMM server.

### CODE REVIEW

File: docs/details/clickhouse.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# ClickHouse
+
+
+You can use an external Clickhouse database instance outside the PMM Server container running on other hosts.
```
Comment (denisok) [inline docs/details/clickhouse.md:4]:
```suggestion
You can use an external ClickHouse database instance outside the PMM Server container running on other hosts.
```

File: docs/details/clickhouse.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ 
+`PERCONA_TEST_PMM_CLICKHOUSE_ADDR` -> hostname:port
+:   Name of the host and port of the external ClickHouse database instance. 
+
+**Optional environment variables**
+
+`PERCONA_TEST_PMM_CLICKHOUSE_DATABASE` -> database name
+:   Database name of the external ClickHouse database instance.
+
+`​​PERCONA_TEST_PMM_CLICKHOUSE_POOL_SIZE` -> [pool size](https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.xml#L1130)
```
Comment (denisok) [inline docs/details/clickhouse.md:20]: is this a right link ? https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.xml#L288 probably?

Should we link to the some commit so it wouldn't change? @nikita-b

File: docs/details/clickhouse.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# ClickHouse
+
+
+You can use an external Clickhouse database instance outside the PMM Server container running on other hosts.
+
+## Environment variables
```
Comment (denisok) [inline docs/details/clickhouse.md:6]: please also add it here: https://github.com/percona/pmm-doc/blob/08a1633b8d86da102ad1dda81557e1489731232f/docs/setting-up/server/docker.md#environment-variables

File: docs/details/index.md [hunk 4]
```diff
@@ -15,6 +15,7 @@
 
 - [API](api.md): How to access the Swagger API.
 
-- [VictoriaMetrics](victoria-metrics.md): The monitoring solution and time-series database that replaced Prometheus in [PMM 2.12.0](../release-notes/2.12.0.md).
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/details/index.md:18]: Reinstated.

**Comment by rnovikovP** [comment]
Can we add a warning that all PERCONA_TEST_* are  Fully experimental and NOT for any production use .. it's internal/development  use only

File: docs/setting-up/server/docker.md [hunk 5]
```diff
@@ -267,25 +267,28 @@ How to run PMM Server with Docker based on our [Docker image].
...
+| `METRICS_RESOLUTION`                                            | High metrics resolution in seconds.
+| `METRICS_RESOLUTION_HR`                                         | High metrics resolution (same as above).
+| `METRICS_RESOLUTION_MR`                                         | Medium metrics resolution in seconds.
+| `METRICS_RESOLUTION_LR`                                         | Low metrics resolution in seconds.
+| `DATA_RETENTION`                                                | How many days to keep time-series data in ClickHouse.
+| `ENABLE_VM_CACHE`                                               | Enable cache in VM.
+| `ENABLE_ALERTING`                                               | Enable integrated alerting.
+| `ENABLE_AZUREDISCOVER`                                          | Enable support for discovery of Azure databases.
+| `ENABLE_BACKUP_MANAGEMENT`                                      | Enable integrated backup tools.
+| `PERCONA_TEST_SAAS_HOST`                                        | SaaS server hostname.
```
Comment (rnovikovP) [inline docs/setting-up/server/docker.md:283]: Let's even create a separate table for `PERCONA_TEST_` vars's and describe them separately ?

**Review by Nailya — CHANGES_REQUESTED** [review]

File: docs/details/clickhouse.md (hunk 3 above)
Comment (rasika-chivate) [reply docs/details/clickhouse.md:6]: done

File: docs/details/clickhouse.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/details/clickhouse.md]: ok

File: docs/setting-up/server/docker.md (hunk 5 above)
Comment (rnovikovP) [reply docs/setting-up/server/docker.md:283]: https://jira.percona.com/browse/PMM-10078?focusedCommentId=297657&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-297657

File: docs/details/clickhouse.md (hunk 2 above)
Comment (nikita-b) [reply docs/details/clickhouse.md:20]: Yes you are right, but I think we'll need to clarify that it's the same setting but for connection: 
```
The maximum number of threads in the current connection thread pool. Can't be bigger then: max_thread_pool_size: https://github.com/ClickHouse/ClickHouse/blob/ba26b3cf4c982f681f7f9782bba2662f97620835/programs/server/config.xml#L288
```

File: docs/details/clickhouse.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ 
+`PERCONA_TEST_PMM_CLICKHOUSE_ADDR` -> hostname:port
+:   Name of the host and port of the external ClickHouse database instance. 
+
+**Optional environment variables**
+
+`PERCONA_TEST_PMM_CLICKHOUSE_DATABASE` -> database name
+:   Database name of the external ClickHouse database instance.
+
+`​​PERCONA_TEST_PMM_CLICKHOUSE_POOL_SIZE` -> [pool size](https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.xml#L1130)
```
Comment (nikita-b) [inline docs/details/clickhouse.md:20]: We have `database name` without square brackets and this and max_block_size with

**Review by nikita-b — CHANGES_REQUESTED** [review]

**Review by denisok — APPROVED** [review]

**Review by rnovikovP — APPROVED** [review]

File: docs/details/index.md [hunk 7]
```diff
@@ -17,4 +17,6 @@
 
 - [VictoriaMetrics](victoria-metrics.md): The monitoring solution and time-series database that replaced Prometheus in [PMM 2.12.0](../release-notes/2.12.0.md).
 
+- [ClickHouse](clickhouse.md):  A third-party column-oriented database that facilitates the Query Analytics functionality.
```
Comment (dygksy) [inline docs/details/index.md:20]:
```suggestion
- [ClickHouse](clickhouse.md):  A third-party column-oriented database management system (DBMS) that facilitates the Query Analytics functionality.
```

**Review by dygksy — APPROVED** [review]

File: docs/setting-up/server/docker.md [hunk 8]
```diff
@@ -267,25 +267,37 @@ How to run PMM Server with Docker based on our [Docker image].
...
+| `METRICS_RESOLUTION`                                            | High metrics resolution in seconds.
+| `METRICS_RESOLUTION_HR`                                         | High metrics resolution (same as above).
+| `METRICS_RESOLUTION_MR`                                         | Medium metrics resolution in seconds.
+| `METRICS_RESOLUTION_LR`                                         | Low metrics resolution in seconds.
+| `DATA_RETENTION`                                                | How many days to keep time-series data in ClickHouse.
+| `ENABLE_VM_CACHE`                                               | Enable cache in VM.
+| `ENABLE_ALERTING`                                               | Enable integrated alerting.
+| `ENABLE_AZUREDISCOVER`                                          | Enable support for discovery of Azure databases.
+| `ENABLE_BACKUP_MANAGEMENT`                                      | Enable integrated backup tools.
+| `ENABLE_DBAAS`                                                  | Enable DBaaS features.
```
Comment (catalinaadam) [inline docs/setting-up/server/docker.md:283]: inconsistency in the description column: Enables/Enable, Disables/Disable.

**Pull request merged**